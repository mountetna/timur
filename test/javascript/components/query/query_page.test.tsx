import React from 'react';
import {Provider} from 'react-redux';
import {
  QueryProvider,
  defaultContext
} from '../../../../lib/client/jsx/contexts/query/query_context';
import {StylesProvider} from '@material-ui/styles/';
import renderer from 'react-test-renderer';
import {mockStore, generateClassName} from '../../helpers';
import QueryPage from '../../../../lib/client/jsx/components/query/query_page';

const models = {
  monster: {template: require('../../fixtures/template_monster.json')},
  labor: {template: require('../../fixtures/template_labor.json')},
  project: {template: require('../../fixtures/template_project.json')}
};

describe('QueryPage', () => {
  let store;

  beforeEach(() => {});

  it('renders', () => {
    store = mockStore({
      magma: {models},
      janus: {projects: require('../../fixtures/project_names.json')}
    });

    global.CONFIG = {
      magma_host: 'https://magma.test'
    };

    // Wrap with Provider here so store gets passed down to child components in Context
    const tree = renderer
      .create(
        <Provider store={store}>
          <StylesProvider generateClassName={generateClassName}>
            <QueryProvider
              state={{
                ...defaultContext.state,
                rootModel: 'monster',
                rootIdentifier: {
                  model_name: 'monster',
                  attribute_name: 'name',
                  display_label: 'monster.name'
                },
                recordFilters: [
                  {
                    modelName: 'labors',
                    attributeName: 'year',
                    operator: '::equals',
                    operand: 2
                  }
                ],
                slices: {
                  prize: [
                    {
                      modelName: 'prize',
                      attributeName: 'name',
                      operator: '::equals',
                      operand: 'Athens'
                    }
                  ]
                }
              }}
            >
              <QueryPage />
            </QueryProvider>
          </StylesProvider>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
