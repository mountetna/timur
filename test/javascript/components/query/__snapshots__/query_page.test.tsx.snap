// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryPage renders 1`] = `
<div
  className="MuiGrid-root query-page MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        >
          <div
            className="MuiGrid-root makeStyles-container MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
          >
            <div
              className="MuiGrid-root makeStyles-container MuiGrid-item"
            >
              <div
                className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  className="MuiCardHeader-root"
                >
                  <div
                    className="MuiCardHeader-content"
                  >
                    <span
                      className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      From
                    </span>
                    <span
                      className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      the root model for your data frame
                    </span>
                  </div>
                </div>
                <div
                  className="MuiCardContent-root"
                >
                  <div
                    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-item"
                    >
                      <div
                        className="MuiFormControl-root makeStyles-formControl"
                      >
                        <label
                          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                          data-shrink={true}
                          id="Root Model-"
                        >
                          Root Model
                        </label>
                        <div
                          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-selectEmpty MuiInputBase-formControl MuiInput-formControl"
                          onClick={[Function]}
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="Root Model-"
                            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                            onBlur={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onMouseDown={[Function]}
                            role="button"
                            tabIndex={0}
                          >
                            <em>
                              None
                            </em>
                          </div>
                          <input
                            aria-hidden={true}
                            className="MuiSelect-nativeInput"
                            onAnimationStart={[Function]}
                            onChange={[Function]}
                            required={false}
                            tabIndex={-1}
                            value=""
                          />
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSelect-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-container MuiGrid-item"
            />
            <div
              className="MuiGrid-root makeStyles-container MuiGrid-item"
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-previewPane MuiGrid-item"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  />
</div>
`;
